File: 271-encode-and-decode-strings/encode-and-decode-strings.py
class Codec:
    def encode(self, strs: List[str]) -> str:
        """Encodes a list of strings to a single string.
        """
        result = ""
        for s in strs:
            result += str(len(s)) + "#" + s
        
        print("result : ", result)
        return result

    def decode(self, s: str) -> List[str]:
        """Decodes a single string to a list of strings.
        """

        strs = []
        i = 0
        while i < len(s):
            j = s.find("#", i)
            print("J : ", j)
            length = int(s[i:j])
            strs.append(s[j+1: j+1+length])
            i = j + 1 + length
        print("result from 2nd :", strs)

        return strs
        
        


# Your Codec object will be instantiated and called as such:
# codec = Codec()
# codec.decode(codec.encode(strs))

File: 271-encode-and-decode-strings/encode-and-decode-strings.java
public class Codec {

    // Encodes a list of strings to a single string.
    public String encode(List<String> strs) {
        StringBuilder result = new StringBuilder();
        for (String s : strs) {
            result.append(s.length()).append("#").append(s);
        }
        return result.toString();
    }

    // Decodes a single string to a list of strings.
    public List<String> decode(String s) {
        List<String> result = new ArrayList<>();
        int i = 0;

        while (i < s.length()) {
            int j = s.indexOf('#', i); // Find the next separator
            int length = Integer.parseInt(s.substring(i, j)); // Extract length
            String str = s.substring(j + 1, j + 1 + length); // Extract the string
            result.add(str);
            i = j + 1 + length; // Move to the next encoded string
        }

        return result;
    }
}